### PR TITLE
fix: preserve the fragment portion (hash) of the URL while redirecting

### LINKED ISSUE #107: When fixing URLs with hashes, hashes are removed
Running `textlint --fix` on

```
http://mochajs.org/#bdd 
```

Results in:

```
https://mochajs.org/
```

### PR DESCRIPTION
**Pull request by 0x6b**
Closes #107.

### CODE REVIEW

File: src/no-dead-link.js [hunk 1]
```diff
@@ -86,7 +86,10 @@ async function isAliveURI(uri, method = 'HEAD') {
       return {
         ok: finalRes.ok,
         redirected: true,
-        redirectTo: finalRes.url,
+        redirectTo:
+          URL.parse(uri).hash !== null
```
Comment (azu) [inline src/no-dead-link.js]: we should re-use `hash` variable.
```js
const hash = URL.parse(uri).hash;
return {
        ok: finalRes.ok,
        redirected: true,
        redirectTo: finalRes.url,
        redirectTo: hash !== null
            ? `${finalRes.url}${hash}`
            : finalRes.url,
       ....
```

Comment (azu) [reply src/no-dead-link.js]: 📝 `hash` propperty will be `null`
https://runkit.com/azu/5ba83e94f20fef001259c176

Comment (0x6b) [reply src/no-dead-link.js]: @azu, thank you for your review. I will add additional commit to fix this, then publish it as 4.5.0.

**Comment by azu** [comment]
Thanks 👍

File: src/no-dead-link.js (hunk 1 above)
Comment (0x6b) [reply src/no-dead-link.js]: Ah this is bug fix so should be patch release.

**Pull request merged**